extensions/sql-migration/src/wizard/targetSelectionPage.ts

```diff
@@ -120,9 +121,14 @@ export class TargetSelectionPage extends MigrationWizardPage {
 			const resourceDropdownValue = (<azdata.CategoryValue>this._azureResourceDropdown.value)?.displayName;
 			switch (this.migrationStateModel._targetType) {
 				case MigrationTargetType.SQLMI: {
-					if (!this.migrationStateModel._targetServerInstance ||
-						resourceDropdownValue === constants.NO_MANAGED_INSTANCE_FOUND) {
+					let targetMi = this.migrationStateModel._targetServerInstance as azureResource.AzureSqlManagedInstance;
+					if (!targetMi || resourceDropdownValue === constants.NO_MANAGED_INSTANCE_FOUND) {
 						errors.push(constants.INVALID_MANAGED_INSTANCE_ERROR);
+						break;
+					}
+					if (targetMi.properties.state !== 'Ready') {
+						errors.push(constants.MI_NOT_READY_ERROR(targetMi.name, targetMi.properties.state));
+						break;
```

Comment:
nit: the `break` is redundant

Reply:
Got it, it's redundant in this second `if` block but not in the first one I believe. Since if we have a null `targetMi` it needs to break immediately or else it'll fall through to the second block and it'll fail because it won't be able to look up the state. So I put in a `break` in each block to make that clear, but I can remove it to clean it up